pgmpy/inference/ApproxInference.py

```diff
@@ -196,3 +202,85 @@ def query(
         return self.get_distribution(
             samples, variables=variables, state_names=state_names, joint=joint
         )
+
+    def map_query(
+        self,
+        variables,
+        n_samples=int(1e4),
+        samples=None,
+        evidence=None,
+        virtual_evidence=None,
+        state_names=None,
+        show_progress=True,
+        seed=None,
+    ):
+        """
+        Computes the MAP Query using approx inference. Returns the
```

Comment:
I think using MAP query in the docstring could be confusing to a lot of users (plus is repeated information as the method name) as they might not know what it means. I would suggest having the first line as: "Finds the highest probable state in the joint distribution of `variables`. Then the next lines can describe how the method computes the values.